server/controllers/user.controller.js

```diff
@@ -0,0 +1,143 @@
+
+exports.create = async (req, res) => {
+    if (!req.body 
+        || !req.body.email
+        || !req.body.password
+        || !req.body.steamId
+    ) {
+        return res.status(400).send({
+            message: "Content cannot be empty."
+        })
+    }
+
+    const user = {
+        email: req.body.email,
+        password: req.body.password,
+        steamId: req.body.steamId
+    }
+
+    await userService.createUser(user)
+    .then(response => {
+        return res.send({
+            message: "User created successfully.",
+            ...response
+        })
+    })
+    .catch(error => {
+        if (error.code) {
+            return res.status(400).send({
+                status: 400,
+                code: error.code,
+                message: error.message
+            })
+        }
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to create the user."
+        })
+    })
+}
+
+exports.findAll = async (req, res) => {
+    hasPermission(res, 3)
+
+    await userService.getAllUsers()
+    .then(response => {
+        return res.send(response)
+    })
+    .catch(error => {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to get all users."
+        })
+    })
+}
+
+exports.find = async (req, res) => {
+    hasPermission(res, 3)
+
+    if (!req.params.id) {
+        return res.status(400).send({
+            message: "Id cannot be empty."
+        })
+    }
+
+    try {
+        let response
+
+        if (req.query.uid)
+            response = await userService.getUserByUID(req.params.id)
+        else
+            response = await userService.getUserById(req.params.id)
+
+        return res.send(response)
+    } catch (error) {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to get the user."
+        })
+    }
+}
+
+exports.modifyUser = async (req, res) => {
+    hasPermission(res, 1)
+
+    if (!req.body 
+        || !req.body.id
+        || !req.body.firebaseUID
+        || !req.body.email
+        || !req.body.steamId
+        || !req.body.role
+        || !req.body.isWhitelisted
+    ) {
+        return res.status(400).send({
+            message: "Content cannot be empty."
+        })
+    }
+
+    const user = {
+        id: req.body.id,
+        firebaseUID: req.body.firebaseUID,
+        email: req.body.email,
+        steamId: req.body.steamId,
+        role: req.body.role,
+        isWhitelisted: req.body.isWhitelisted
+    }
+
+    await userService.updateUser(user)
+    .then(response => {
+        return res.send(response)
+    }) 
+    .catch(error => {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to modify the user."
+        })
+    })
+}
```

Comment:
Since this hasPermission 1 (admin) but I want managers to be able to modify the whitelist setting, either add a method to handle only whitelist change or add a condition in this method. Also, the user will be able to manage his own account eventually so maybe a condition could be the better solution here since we would have 3 different ways of calling the same method with 3 different permission levels.

Reply:
Still wondering if I want managers to be able to modify the steamId and email or a viewer user.
also a manager should not be able to change another manger's info

Reply:
Restriction has been added in the front and backend in commit [c871d98](https://github.com/Funnyadd/assetto-corsa-server-dashboard/pull/1/commits/c871d98541b707afad9b6b78b31f929b7268570e)